### ISSUE TITLE
Netcdf Trajectory Not Loading

### ISSUE THREAD

**Issue by pd15ms089** [MDAnalysis/mdanalysis#3145]
## Expected behavior ##

I have a Netcdf file which was constructed by appending several Netcdf trajectories using cpptraj. With VMD there is no issue reported while loading the trajectory. Ideally MDAnalysis also should work. My code while using MDAnalysis to load the trajectory is following:
```python
import MDAnalysis as mda
u = mda.Universe('my.prmtop','my.nc')
```
As per my understanding with MDAnalysis, It should load the trajectory without any warning.

## Actual behavior ##

However, I found an ValueError which shows the following error code:
```python
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-eb338ed31b92> in <module>
----> 1 u=mda.Universe('my.prmtop','my.nc')

~/anaconda3/lib/python3.8/site-packages/MDAnalysis/core/universe.py in __init__(self, *args, **kwargs)
    387 
    388         if coordinates:
--> 389             self.load_new(coordinates, format=format, in_memory=in_memory,
    390                         in_memory_step=in_memory_step, **kwargs)
    391 

~/anaconda3/lib/python3.8/site-packages/MDAnalysis/core/universe.py in load_new(self, filename, format, in_memory, in_memory_step, **kwargs)
    580         kwargs['n_atoms'] = self.atoms.n_atoms
    581 
--> 582         self.trajectory = reader(filename, format=format, **kwargs)
    583         if self.trajectory.n_atoms != len(self.atoms):
    584             raise ValueError("The topology and {form} trajectory files don't"

~/anaconda3/lib/python3.8/site-packages/MDAnalysis/coordinates/TRJ.py in __init__(self, filename, n_atoms, mmap, **kwargs)
    474         super(NCDFReader, self).__init__(filename, **kwargs)
    475 
--> 476         self.trjfile = scipy.io.netcdf.netcdf_file(self.filename,
    477                                                    mmap=self._mmap)
    478 

~/anaconda3/lib/python3.8/site-packages/scipy/io/netcdf.py in __init__(self, filename, mode, mmap, version, maskandscale)
    279 
    280         if mode in 'ra':
--> 281             self._read()
    282 
    283     def __setattr__(self, attr, value):

~/anaconda3/lib/python3.8/site-packages/scipy/io/netcdf.py in _read(self)
    608         self._read_dim_array()
    609         self._read_gatt_array()
--> 610         self._read_var_array()
    611 
    612     def _read_numrecs(self):

~/anaconda3/lib/python3.8/site-packages/scipy/io/netcdf.py in _read_var_array(self)
    714             # Build rec array.
    715             if self.use_mmap:
--> 716                 rec_array = self._mm_buf[begin:begin+self._recs*self._recsize].view(dtype=dtypes)
    717                 rec_array.shape = (self._recs,)
    718             else:

ValueError: When changing to a larger dtype, its size must be a divisor of the total size in bytes of the last axis of the array.
```

## Code to reproduce the behavior ##

The above behavior can be reproduced with:
```python
import MDAnalysis as mda
u = mda.Universe('my.prmtop','my.nc')
```
However the trajectory used here were constructed by appending several netcdf trajectories one after another.

## Current version of MDAnalysis ##

- MDAnalysis version: 1.0.0
- Python 3.8.3
- Ubuntu 20.04.2 LTS (64 bit)

**Comment by IAlibay** [comment]
Thanks for reporting this issue @pd15ms089 would you be able to upload an example trajectory ? (ideally one that we can host here under GPLv2). It's not immediately clear to me as to what is happening here, I suspect our readers are expecting a different float size than what VMD is storing values as.

**Comment by orbeckst** [comment]
Can you open your nc file with

```python
import scipy.io.netcdf

f = scipy.io.netcdf.netcdf_file("my.nc")
```

?
